sdk/cosmos/azure-cosmos/azure/cosmos/aio/container.py

```diff
@@ -0,0 +1,714 @@
+        """Get the item identified by `item`.
+
+        :param item: The ID (name) or dict representing item to retrieve.
+        :param partition_key: Partition key for the item to retrieve.
+        :keyword str session_token: Token for use with Session consistency.
+        :keyword dict[str,str] initial_headers: Initial headers to be sent as part of the request.
+        :keyword Callable response_hook: A callable invoked with the response metadata.
+        :returns: Dict representing the item to be retrieved.
+        :raises ~azure.cosmos.exceptions.CosmosHttpResponseError: The given item couldn't be retrieved.
+        :rtype: dict[str, Any]
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/examples_async.py
+                :start-after: [START update_item]
+                :end-before: [END update_item]
+                :language: python
+                :dedent: 0
+                :caption: Get an item from the database and update one of its properties:
+                :name: update_item
+        """
+        doc_link = self._get_document_link(item)
+        request_options = build_options(kwargs)
+        response_hook = kwargs.pop('response_hook', None)
+        if partition_key is not None:
+            request_options["partitionKey"] = await self._set_partition_key(partition_key)
+
+        result = await self.client_connection.ReadItem(document_link=doc_link, options=request_options, **kwargs)
+        if response_hook:
+            response_hook(self.client_connection.last_response_headers, result)
+        return result
+
+    @distributed_trace
+    def read_all_items(
+        self,
+        max_item_count=None,  # type: Optional[int]
+        **kwargs  # type: Any
+    ):
+        # type: (...) -> Iterable[Dict[str, Any]]
+        """List all the items in the container.
+
+        :param max_item_count: Max number of items to be returned in the enumeration operation.
+        :keyword str session_token: Token for use with Session consistency.
+        :keyword dict[str,str] initial_headers: Initial headers to be sent as part of the request.
+        :keyword Callable response_hook: A callable invoked with the response metadata.
+        :returns: An Iterable of items (dicts).
+        :rtype: Iterable[dict[str, Any]]
+        """
+        feed_options = build_options(kwargs)
+        response_hook = kwargs.pop('response_hook', None)
+        if max_item_count is not None:
+            feed_options["maxItemCount"] = max_item_count
+
+        if hasattr(response_hook, "clear"):
+            response_hook.clear()
+
+        items = self.client_connection.ReadItems(
+            collection_link=self.container_link, feed_options=feed_options, response_hook=response_hook, **kwargs
+        )
+        if response_hook:
+            response_hook(self.client_connection.last_response_headers, items)
+        return items
+
+    @distributed_trace_async
+    async def query_items(
```

Comment:
we should look into if our async io implementation truly will parallelize the cross partition query execution? or if the execution is serial.

please test and check.